test/tools/cli/IonProcess.java

```diff
@@ -544,14 +502,28 @@ private static Event ionStreamToEvent(IonReader ionReader) {
                     event.setDepth(ionReader.intValue());
                     break;
             }
-        }
 
+        }
+        //compare text value and binary value
+        if (textValue != null && binaryValue != null) {
```

Comment:
If this branch isn't taken, throw an exception, as the event stream is illegal.

Reply:
Wrote a valid function to check each cases.